onnxruntime/core/providers/cpu/nn/conv_attributes.h

```diff
@@ -118,20 +118,133 @@ struct ConvAttributes {
+  }
+
+  // Use this method when pads are to be made symmetrical (if they are asymmetric)
+  // and to collect metadata regarding the portion of the output (with "adjusted" pads)
+  // to be sliced off to make the output correspond to the "actual" asymmetric paddings
+  Status InferOutputShapeWithAdjustedPads(const TensorShape& input_shape,
+                                          const std::vector<int64_t>& kernel_shape,
+                                          const std::vector<int64_t>& strides_p,
+                                          const std::vector<int64_t>& dilations_p,
+                                          std::vector<int64_t>& pads_p,
+                                          std::vector<int64_t>& output_shape,
+                                          std::vector<int64_t>& output_shape_with_revised_pads,
+                                          bool& post_slicing_needed,
+                                          std::vector<int64_t>& slice_starts,
+                                          std::vector<int64_t>& slice_ends,
+                                          std::vector<int64_t>& slice_axes) const {
+    size_t rank = input_shape.NumDimensions();
+    for (size_t dim = 0; dim < rank; ++dim) {
+      if (dim >= strides_p.size() || dim >= kernel_shape.size() ||
+          dim >= dilations_p.size() || dim >= pads_p.size() ||
+          rank + dim >= pads_p.size()) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Out of bound access to array");
+      }
+      int64_t output_dim_size = 0;
+      ORT_RETURN_IF_ERROR(ComputePadAndOutputShape(input_shape[dim],
+                                                   strides_p[dim],
+                                                   kernel_shape[dim],
+                                                   dilations_p[dim],
+                                                   auto_pad,
+                                                   pads_p[dim],
+                                                   pads_p[rank + dim],
+                                                   output_dim_size));
+      if (output_dim_size <= 0) {
+        return Status(common::ONNXRUNTIME, common::INVALID_ARGUMENT, "Invalid input shape: " + input_shape.ToString());
+      }
+
+      // This is the "actual" output shape of the Conv op (i.e.) with given pads as is
+      output_shape.push_back(output_dim_size);
+
+      // Deal with asymmetric pads if any and adjust them to be symmetric
+      // Along the way - note down how many values need to be sliced out from the start and end
+      // of each spatial dimension while we over-pad to get symmetric pads
+
+      if (pads_p[dim] == pads_p[rank + dim]) {
+        // symmetric padding - No operation as such
+        // Make note of the dim size of the output (to be used if there are other symmetrically padded dims)
+        output_shape_with_revised_pads.push_back(output_dim_size);
+      } else {
+        // asymmetric padding
```

Comment:
deal with asymmetric padding and "adjust" them to be symmetric and hold onto metadata concerning which parts of the result must be "sliced" out